detox/src/devices/Device.js

```diff
@@ -3,15 +3,26 @@ const path = require('path');
 const _ = require('lodash');
 const argparse = require('../utils/argparse');
 const configuration = require('../configuration');
+const sh = require('../utils/sh');
+const log = require('npmlog');
 
 class Device {
 
   constructor(deviceConfig, sessionConfig, deviceDriver) {
     this._deviceConfig = deviceConfig;
     this._sessionConfig = sessionConfig;
     this.deviceDriver = deviceDriver;
+    this._currentLaunchNumber = 0;
+    this._currentTestArtifactsDestination = undefined;
 
     this.deviceDriver.validateDeviceConfig(deviceConfig);
+
+    this.relaunchApp = this.relaunchApp.bind(this);
```

Comment:
I probably don't need all of them, only those whose this is incorrect and is called from outside this. I guess I can reduce the list, but some will have to stay there